src/StreamingDataFrames/documentation/StreamingDataFrame.md

```diff
@@ -0,0 +1,236 @@
+
+# `StreamingDataFrame`: Detailed Overview
+
+Now that we've outlined how to get an `Application` up and running, what exactly can you
+do with a `StreamingDataFrame`?
+
+<br>
+
+For example purposes, assume the record we are manipulating looks like this:
+
+```python
+row = {
+    "field_a": "my_str", 
+    "field_b": "my_other_str",
+    "field_c": [1, 2, 3],
+    "field_d": "DELETE ME",
+}
+```
+
+Along with a `StreamingDataFrame` instance, as generated by an `Application`:
+
+```python
+sdf = Application().dataframe()
+```
+
+<br>
+
+## Interacting with `Rows`
+
+Under the hood, `StreamingDataFrame` is manipulating kafka messages via `Row` objects.
+
+Simplified, a `Row` is effectively a dictionary of the Kafka message 
+value, with each key equivalent to a dataframe column name. 
+
+Our `StreamingDataFrame` interacts with `Row` objects via the Pandas dataframe
+interface, unless specified otherwise (i.e. the `.apply()` feature).
+
+As a user, their existence should largely go unnoticed.
+
+
+<br>
+
+## Accessing Fields/Columns
+
+In typical Pandas dataframe fashion, you can access a column:
+
+```python
+sdf["field_a"]  # "my_str"
+```
+
+Typically, this is done in combination with other operations.
+
+You can also access nested objects (dicts, lists, etc):
+
+```python
+sdf["field_c"][2]  # 3
+```
+
+
+<br>
+
+## Performing Operations with Columns
+
+In typical Pandas dataframe fashion, you can do almost any basic operations or 
+comparisons with columns, assuming validity:
+
+```python
+sdf["field_a"] + sdf["field_b"]
+sdf["field_a"] or sdf["field_b"]
+sdf["field_a"] & sdf["field_b"]
+sdf["field_a"] is not None
+sdf["field_a"] != "woo"
+```
+
+
+<br>
+
+## Assigning New Columns
+
+In typical Pandas fashion, you can add new columns from the results of numerous other
+operations:
+
+```python
+sdf["a_new_int_field"] = 5 
+sdf["a_new_str_field"] = sdf["field_a"] + sdf["field_b"]
+sdf["another_new_field"] = sdf["a_new_str_field"].apply(lambda value: value + "another")
+```
+
+See [the `.apply()` section](#user-defined-functions-apply) for more information on how that works.
+
+
+<br>
+
+## Subsetting/Slicing Columns
+
+In typical Pandas fashion, you can take a subset of columns:
+
+```python
+# remove "field_d"
+sdf = sdf[["field_a", "field_b", "field_c"]]
```

Comment:
Instead of `remove "filed_d"` I would say
`removes all fields except for field_a, field_b and field_c`
Because it requires less context to understand